rio/commands/detectTag.py

```diff
@@ -0,0 +1,18 @@
+import apriltag
+import cv2 as cv
+import commands2
+
+# is this where this file should go?
+class detectTag(commands2.CommandBase):
+    def __init__(self) -> None:
+        super().__init__()
+        self.cam = cv.VideoCapture(0)
```

Comment:
This is too much load for the rio to handle, see my earlier comment about offloading RIO tasks away from the main code. Running this may cause the rio to lockup and under-run

Reply:
I am planning on putting it on the raspberry pi but am not sure how

Reply:
Make a new folder for the raspberry pi code, you can use the wpilib vision image or just run your own. See last years dashboard for an example: https://github.com/FRC-1721/1721-RapidReact/tree/main/dashboard

it uses a [makefile to sftp all the code over to the coprocessor ](https://github.com/FRC-1721/1721-RapidReact/blob/a8ba43fd3ed713b8fc7e0688153e1a2970b03d00/dashboard/Makefile#L40-L55)

Something like that would work very well! Then just use networktables to retrieve the data back on the rio side. Just like how you would with the limelight

Speaking of the limelight, the new limelight does this for you now! basically the exact same thing, it even tracks april tags and has docs: https://limelightvision.io/

Maybe ask mat if you can buy a new one!